iclr_wrap_up/utils.py

```diff
@@ -1,102 +1,46 @@
-    X is the training data
-    y is training class, with numbers from 0 to 1
-    Y is training class, but coded as a 2-dim vector with one entry set to 1 at the column index corresponding to the class
+    Returns:
+        A new Namedtuple with fields X, y and Y containing the concatenation of training and test data
     """
-    nb_classes = 2
-    data_file = Path('datasets/IB_data_'+str(ID)+'.npz')
-    if data_file.is_file():
-        data = np.load('datasets/IB_data_'+str(ID)+'.npz')
-    else:
-        create_IB_data(ID)
-        data = np.load('datasets/IB_data_'+str(ID)+'.npz')
-        
-    (X_train, y_train), (X_test, y_test) = (data['X_train'], data['y_train']), (data['X_test'], data['y_test'])
-
-    Y_train = utils.to_categorical(y_train, nb_classes).astype('float32')
-    Y_test  = utils.to_categorical(y_test, nb_classes).astype('float32')
-
     Dataset = namedtuple('Dataset',['X','Y','y','nb_classes'])
-    trn = Dataset(X_train, Y_train, y_train, nb_classes)
-    tst = Dataset(X_test , Y_test, y_test, nb_classes)
-    del X_train, X_test, Y_train, Y_test, y_train, y_test
-    return trn, tst
+    X = np.concatenate((training.X, test.X))
+    y = np.concatenate((training.y, test.y))
+    Y = np.concatenate((training.Y, test.Y))
+    return Dataset(X, Y, y, training.nb_classes)
 
-def create_IB_data(idx):
-    data_sets_org = load_data()
-    data_sets = data_shuffle(data_sets_org, 80, shuffle_data=True)
-    X_train, y_train, X_test, y_test = data_sets.train.data, data_sets.train.labels[:,0], data_sets.test.data, data_sets.test.labels[:,0]
-    np.savez_compressed('datasets/IB_data_'+str(idx), X_train=X_train, y_train=y_train, X_test=X_test, y_test=y_test)
-
-def construct_full_dataset(trn, tst):
-    Dataset = namedtuple('Dataset',['X','Y','y','nb_classes'])
-    X = np.concatenate((trn.X,tst.X))
-    y = np.concatenate((trn.y,tst.y))
-    Y = np.concatenate((trn.Y,tst.Y))
-    return Dataset(X, Y, y, trn.nb_classes)
- 
-def load_data():
-    """Load the data
-    name - the name of the dataset
-    return object with data and labels"""
-    print ('Loading Data...')
-    C = type('type_C', (object,), {})
-    data_sets = C()
-    d = sio.loadmat('datasets/var_u.mat')
-    F = d['F']
-    y = d['y']
-    C = type('type_C', (object,), {})
-    data_sets = C()
-    data_sets.data = F
-    data_sets.labels = np.squeeze(np.concatenate((y[None, :], 1 - y[None, :]), axis=0).T)
-    return data_sets
 
 def shuffle_in_unison_inplace(a, b):
-    """ Shuffle the arrays randomly """
+    """ Shuffles both array a and b randomly in unison
+    Args:
+        a: An Array, for example containing data samples
+        b: An Array, fpor example containing labels
+
+    Returns:
+        Both arrays shuffled in the same way
+    """
     assert len(a) == len(b)
     p = np.random.permutation(len(a))
     return a[p], b[p]
 
+
 def data_shuffle(data_sets_org, percent_of_train, min_test_data=80, shuffle_data=False):
```

Comment:
This code is a little opaque to me. Maybe this could be replaced with a preimplemented method from sklearn like http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html